From 555baabd1c8d44cd3be0aa8676a535881cfcd384 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Fri, 23 Oct 2015 16:21:18 +0800
Subject: [PATCH] Delete VCM power on platform Brightonshores

There is no VCM in Brightonshores. So we delete VCM power on platform
Brightonshores.

Change-Id: I3c50c621c14086d8480f986e36944857e21c3904
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-8243
Original-Tracked-On: #H1504076511
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/81375
Approver: Bin B Han <bin.b.han@intel.com>
Build: Dessert M COHO (SC)
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: Dessert M BXT RVP (SC)
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 4 ++++
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.h | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 9089132..393a41e 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -1672,7 +1672,11 @@ static int __update_ov8858_device_settings(struct ov8858_device *dev,
 					   u16 sensor_id)
 {
 	if (sensor_id == OV8858_CHIP_ID)
+#ifdef CONFIG_PLATFORM_BTNS
+		dev->vcm_driver = &ov8858_vcms[OV8858_ID_DEFAULT];
+#else
 		dev->vcm_driver = &ov8858_vcms[OV8858_SUNNY];
+#endif
 	else
 		return -ENODEV;
 
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
index 53a7c35..13c8fb1 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
@@ -337,8 +337,8 @@ static struct ov8858_vcm ov8858_vcms[] = {
 		.t_vcm_timing = dw9718_t_vcm_timing,
 	},
 	[OV8858_ID_DEFAULT] = {
-		.power_up = vcm_power_up,
-		.power_down = vcm_power_down,
+		.power_up = NULL,
+		.power_down = NULL,
 	},
 };
 
-- 
1.9.1

